File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    
    // first solution
    
/*int majorityElement(vector<int>& nums){
    int el = nums[0];
    int count = 0;
    for(int i = 0; i<nums.size(); i++){
        if(count == 0) 
            el = nums[i];
        count+=(el==nums[i])?1:-1;
    }
    return el;
}*/
    
    // second solution
    
int majorityElement(vector<int>& nums){
    
    unordered_map<int,int> mp;
    
    for(int i=0; i<nums.size(); i++){
        mp[nums[i]]++;
    }
    
    for(auto el : mp){
        if(el.second > nums.size()/2){
            return el.first;
        }
    }
    return -1;
}    
 
};

File: 0169-majority-element/0169-majority-element.java
import java.util.Hashtable;
import java.util.Map;
class Solution {
    public int majorityElement(int[] nums) {
        Hashtable<Integer, Integer> tb = new Hashtable<>();

        for(int i=0; i<nums.length; i++){
            if(tb.containsKey(nums[i])){
                int val = tb.get(nums[i]);
                tb.replace(nums[i],val, ++val);
            }
            else{
                tb.put(nums[i],1);
            }
        }
        int res = 0;
        for(Map.Entry<Integer, Integer> el : tb.entrySet()){
            if(el.getValue() > nums.length/2){
                res = el.getKey();
            }
        }
        return res;
    }
}